js/src/indexation.js

```diff
@@ -1,33 +1,123 @@
+ *
+ * This action is executed before the endpoint is first called with the indexing
+ * settings as its first argument.
+ *
+ * @param {string}   endpoint The endpoint on which to register the action.
+ * @param {function} action   The action to register.
+ *
+ * @returns {void}
+ */
+window.yoast.indexing.registerPreIndexingAction = ( endpoint, action ) => {
+	preIndexingActions[ endpoint ] = action;
+};
+
+/**
+ * Registers an action on the given indexing endpoint.
+ *
+ * This action is executed each time after the endpoint is called, with the objects
+ * returned from the endpoint as its first argument and the indexing settings as its second argument.
+ *
+ * @param {string}                       endpoint The endpoint on which to register the action.
+ * @param {function(Object[], Object[])} action   The action to register.
+ *
+ * @returns {void}
+ */
+window.yoast.indexing.registerIndexingAction = ( endpoint, action ) => {
+	IndexingActions[ endpoint ] = action;
+};
+
+/**
+ * @deprecated Deprecated since 15.1. Use `window.yoast.indexing` instead.
+ */
+window.yoast.indexation = window.yoast.indexing;
+
+/**
+ * Registers a pre-indexing action on the given indexing endpoint.
+ *
+ * This action is executed before the endpoint is first called with the indexing
+ * settings as its first argument.
+ *
+ * @deprecated Deprecated since 15.1. Use `window.yoast.indexing.registerPreIndexingAction` instead.
+ *
+ * @param {string}   endpoint The endpoint on which to register the action.
+ * @param {function} action   The action to register.
+ *
+ * @returns {void}
+ */
+window.yoast.indexation.registerPreIndexationAction = ( endpoint, action ) => {
+	console.warn( "Deprecated since 15.1. Use 'window.yoast.indexing.registerPreIndexingAction' instead." );
+	window.yoast.indexing.registerPreIndexingAction( endpoint, action );
+};
+
+/**
+ * Registers an action on the given indexing endpoint.
+ *
+ * This action is executed each time after the endpoint is called, with the objects
+ * returned from the endpoint as its first argument and the indexing settings as its second argument.
+ *
+ * @deprecated Deprecated since 15.1. Use `window.yoast.indexing.registerIndexingAction` instead.
+ *
+ * @param {string}   endpoint The endpoint on which to register the action.
+ * @param {function(Object[], Object[])} action   The action to register.
+ *
+ * @returns {void}
+ */
+window.yoast.indexation.registerIndexationAction = ( endpoint, action ) => {
+	console.warn( "Deprecated since 15.1. Use 'window.yoast.indexing.registerIndexingAction' instead." );
+	window.yoast.indexing.registerIndexingAction( endpoint, action );
+};
+
+/**
+ * Indexes the site and shows a progress bar indicating the indexing process' progress.
+ */
+class Indexing extends Component {
+	/**
+	 * Indexing constructor.
+	 *
+	 * @param {Object} props The properties.
+	 */
+	constructor( props ) {
+		super( props );
+
+		this.settings = yoastIndexingData;
+		this.stoppedIndexing = false;
```

Comment:
This isn't used anywhere?

Reply:
This was mistakenly left over from a previous try of implementing the indexing stop button. It should have been removed. Nice catch!